src/components/Extensions/Details/DetailCard/Tabs/EventsConfig.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Typography } from '@rmwc/typography';
+
+import { Accordion } from '../../../../common/Accordion';
+import { Markdown } from '../../../../common/Markdown';
+import { useExtension } from '../../../api/useExtension';
+import { ParamType } from '../../../models';
+import styles from './EventsConfig.module.scss';
+import { ParamValue } from './ParamValue';
+
+export function EventsConfig() {
+  const extension = useExtension()!;
+  const eventsByPrefix: Map<string, string[]> = new Map(); // maps prefix ==> list of types
+  const eventDescriptionMapping: Map<string, string> = new Map(); // maps type ==> description
+  let allowedEventTypesDiv;
+  let channelLocationDiv;
+  !!extension.events &&
+    extension.events.forEach((event) => {
+      eventDescriptionMapping.set(event.type, event.description);
+    });
+  if (!!extension.allowedEventTypes && extension.allowedEventTypes.length > 0) {
```

Comment:
Same thing wrt to null-chaining

Reply:
Pardon my newbness on this syntax...
Is the correct way to write this:

`if (extension.allowedEventTypes?.length > 0) { ... }`

That's what I tried and it gives me a typescript error that extension.allowedEventTypes could be undefined.